File: api/src/main/java/com/knowledgebase/api/domain/entity/Document.java
package com.knowledgebase.api.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(name = "documents")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Document extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @Column(name = "original_name", nullable = false)
    private String originalName;

    @Column(name = "file_type", nullable = false, length = 100)
    private String fileType;

    @Column(name = "file_size_bytes", nullable = false)
    private Long fileSizeBytes;

    @Column(name = "object_key", nullable = false, unique = true)
    private String objectKey;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by", nullable = false)
    private User uploadedBy;

    @Enumerated(EnumType.STRING)
    @Column(name = "indexing_status", nullable = false, length = 20)
    @Builder.Default
    private com.knowledgebase.api.domain.enums.DocumentIndexingStatus indexingStatus = com.knowledgebase.api.domain.enums.DocumentIndexingStatus.PENDING;

    @Column(name = "chunk_count", nullable = false)
    @Builder.Default
    private Integer chunkCount = 0;

    @Column(name = "indexed_at")
    private java.time.LocalDateTime indexedAt;
}

File: web/src/types/document.ts
export type DocumentIndexingStatus = 'PENDING' | 'PROCESSING' | 'INDEXED' | 'FAILED';

export interface DocumentItem {
  id: string;
  projectId: string;
  originalName: string;
  fileType: string;
  fileSizeBytes: number;
  indexingStatus?: DocumentIndexingStatus;
  chunkCount?: number;
  indexedAt?: string | null;
  uploadedBy: {
    id: string;
    email: string;
    fullName: string;
  } | null;
  createdAt: string;
}

export interface PresignedUrlData {
  url: string;
  expiresInSeconds: number;
}
